pga2d/objects.js

```diff
@@ -0,0 +1,80 @@
+import { Even } from "./multivectors.js";
+
+export class Point {
+  constructor(x, y) {
+    // a euclidean point is represented in 2D PGA as a bivector
+    // normalized so xy = 1, i.e.
+    // a*yo - b*xo + xy
+    // here a is the x-component and b is the y-component. Note the negative
+    // sign, this is due to storing the blades in lexicographic order
+    this.bivec = new Even(0, 1, -y, x);
+  }
+
+  get x() {
+    return this.bivec.yo;
+  }
+
+  get y() {
+    return -this.bivec.xo;
+  }
+
+  add(direction) {
+    const { xo, yo } = this.bivec.add(direction.bivec);
+    return new Point(yo, -xo);
+  }
+
+  sub(point) {
+    const { xo, yo } = this.bivec.sub(point.bivec);
+    return new Direction(yo, -xo);
+  }
+
+  dist_sqr(point) {
+    return this.sub(point).norm();
+  }
+}
+Point.ORIGIN = Object.freeze(new Point(0, 0));
+
+export class Direction {
```

Comment:
I really want to express `Point = Euclidean(x, y) | Ideal(dx, dy)` (or perhaps `PGAPoint = Point(x, y) | Direction(dx, dy)`